### DISCUSSION TITLE
HashSet<Point> is slower than HashSet<String>

### DISCUSSION THREAD

**Discussion by ericwood8** [dotnet/runtime#108306] [Ideas] [answered]
Summary
Fix the Point datatype to be one of the generics throughout all generics in C#.

Motivation
GOAL 1- Performance.
GOAL 2- Less code by developer.

Planning:

    Point needs to be considered one of the generics. So check not only HashSet but other things you can set to generics.
    PointComparer : IEqualityComparer needs a good Equals() and GetHashCode()

Suggested reading:
https://stackoverflow.com/questions/46142734/why-is-hashsetpoint-so-much-slower-than-hashsetstring

**Comment by huoyaoyuan** [comment, marked as answer]
The stackoverflow post is highly outdated. `System.Drawing.Point` implemented `IEquatable` in .NET Core 2.0, which was published at 2017.

**Reply by huoyaoyuan** [reply]
It's also answered at https://github.com/dotnet/csharplang/discussions/8458 with the exact same answer.

**Comment by kasperk81** [comment]
net framework 4.7 https://dotnetfiddle.net/xu6SYJ 0.96s
net8.0 https://dotnetfiddle.net/dy9R9P 0.03s

that's 32 times faster or 97% improvement!
